src/pages/chat/Chat.tsx

```diff
@@ -141,6 +153,88 @@ export default function Chat() {
   const [previousPath, setPreviousPath] = useState<string>("");
   const navigate = useKeepQueryParamsNavigate();
   const { isXXS, isXS, isS } = useBreakpoints();
+  const { showModal, modalTypes } = useModal();
+  const { bosonXmtp } = useChatContext();
+  const notifiConfig = getNotifiConfig();
+
+  // TODO: ideally if notifiConfig is null, do not call useNotifiClient()
+
+  const notifiClient = useNotifiClient({
+    dappAddress: notifiConfig?.dappId || "",
+    env: "Development",
+    walletPublicKey: address as string,
+    walletBlockchain: notifiConfig?.chain || "ETHEREUM"
+  });
+  const fetchNotifiData = (data: NotifiClientData | null) => {
+    const xmtpAlert = data?.alerts?.find(
+      (alert: { filter?: { filterType?: string } }) =>
+        alert.filter?.filterType === "WEB3_CHAT_MESSAGES"
+    );
+    const xmtpSources = xmtpAlert?.sourceGroup?.sources?.filter(
+      (source: unknown) => (source as { type?: string }).type === "XMTP"
+    );
+    const sourceTopics = xmtpSources?.map(
+      (source: unknown) => (source as { name?: string }).name
+    ) as string[];
+    setAlreadyRegisteredTopics(sourceTopics);
+  };
+
+  const onNotifiPopupClosed = useCallback(() => {
+    if (notifiClient) {
+      notifiClient.fetchData(true).then((data) => {
+        fetchNotifiData(data);
+      });
+    }
+  }, [notifiClient]);
+
+  const [alreadyRegisteredTopics, setAlreadyRegisteredTopics] = useState<
+    string[]
+  >([]);
+
+  useEffect(() => {
+    if (!notifiClient || notifiClient.loading) {
+      setAlreadyRegisteredTopics([]);
+    } else {
+      fetchNotifiData(notifiClient.data);
+    }
+  }, [notifiClient]);
```

Comment:
not sure if this causes an infinite rerender loop. I made all the suggested changes and got it but maybe it was there already. I also tried with this and it disappeared but I dont know if it's correct

```
  const existsNotifiClient = !!notifiClient;
  useEffect(() => {
    if (!notifiClient || notifiClient.loading) {
      setAlreadyRegisteredTopics([]);
    } else if (notifiClient.data) {
      fetchNotifiData(notifiClient.data);
    }
  }, [existsNotifiClient]);

```

Reply:
I didn't notice any issue. 
Changing the code as suggested will change the logic (in case `notifiClient.loading`, the `useEffect()` hook will assign `alreadyRegisteredTopics` to [], then when loading is over, there will be no update)